devicetypes/smartthings/zwave-door-window-sensor.src/zwave-door-window-sensor.groovy

```diff
@@ -233,10 +233,17 @@ def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerS
 
 	if (msr == "011A-0601-0901") {  // Enerwave motion doesn't always get the associationSet that the hub sends on join
 		result << response(zwave.associationV1.associationSet(groupingIdentifier: 1, nodeId: zwaveHubNodeId))
-	} else if (!device.currentState("battery")) {
-		if (msr == "0086-0102-0059") {
-			result << response(zwave.securityV1.securityMessageEncapsulation().encapsulate(zwave.batteryV1.batteryGet()).format())
-		} else {
+	} else if (msr == "0086-0102-0059") {
+		def cmds = []
+		if (!device.currentState("battery")) {
+			cmds << zwave.securityV1.securityMessageEncapsulation().encapsulate(zwave.batteryV1.batteryGet()).format()
```

Comment:
I will try first two suggestions - it should be better to generalize this code a little. I am a bit afraid it can break this DTH for other devices that use it, as we do not have all the devices that use this.

Initial poll would not fix issue ICP-4701 by itself, as we already send something similar in configure(), and apparently do not get any response (although this may be caused by wakeUpNoMoreInformation from updated arriving before configure() is called), but I agree updated() seems a bit weird right now.

Now that I think about it, this code should not be in updated() in the first place, as those devices are most likely sleepy, so when we save preferences in Classic, those commands will most likely be sent while the device is asleep. So installed, which is called once during inclusion seems more logical choice.

I think adding battery check to configure could be good, if not for this:
[createEvent(map), response(zwave.wakeUpV1.wakeUpNoMoreInformation())]
in response to battery report. So this 20s delay before wakeUpNoMoreInformation should be better. I will have to check if this is used somewhere else to end communication.

Thanks for input.

Reply:
I have a question. This device seems to handle wakeUpNoMoreInformation command properly only if it was security encapsulated. Should I make a modification like this for all secure devices here, or should I make a check if cc 84 is mentioned in zwInfo.sec first?

Reply:
I think if you replace the `if (state.sec == 1)` with `if ((zwaveInfo.zw == null && state.sec != 0) || zwaveInfo.zw?.endsWIth("s"))` in command() and use either command() or commands() that would probably be safe.

Reply:
@mckeed  Might also have some input.